src/core.jl

```diff
@@ -240,9 +240,9 @@ function AxisArray(A::AbstractArray{T,N}, names::NTuple{N,Symbol}, steps::NTuple
 end
 
 # Alternative constructor, takes names as keywords:
-AxisArray(A; kw...) = AxisArray(A, nt_to_axes(kw.data)...)
-@generated nt_to_axes(nt::NamedTuple) =
-    Expr(:tuple, (:(Axis{$(QuoteNode(n))}(getfield(nt, $(QuoteNode(n))))) for n in nt.names)...)
+AxisArray(A; kw...) = AxisArray(A, nt_to_axes(values(kw)))
+@generated nt_to_axes(nt::NamedTuple{names,T}) where {names,T} =
```

Comment:
Since `T` is not used you could also write

```suggestion
@generated nt_to_axes(nt::NamedTuple{names}) where {names} =
```

Reply:
cool, thanks!